### PR TITLE
feat: add ADR for AdditionalMetadata fields validation

### PR DESCRIPTION
**Pull request by zawan-ila**
# [PROD-3262](https://2u-internal.atlassian.net/browse/PROD-3262)

## Description

This PR adds the ADR for AdditionalMetadata fields validation

### CODE REVIEW

File: decisions/0005-product-source-based-required-fields.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+--------------------------------------------------------------------
+
+Status
+------
+Accepted (Feb 2023)
+
+Context
+-------
+During the recent influx of external products into the marketplace, a number of products were ingested into publisher which did
+not have all the AdditionalMetadata fields associated to them. As such our current publisher flow, which requires these fields
```
Comment (AfaqShuaib09) [inline decisions/0005-product-source-based-required-fields.rst:11]: nit: `associated with`*

File: decisions/0005-product-source-based-required-fields.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Context
+-------
+During the recent influx of external products into the marketplace, a number of products were ingested into publisher which did
+not have all the AdditionalMetadata fields associated to them. As such our current publisher flow, which requires these fields 
+to be present, needed to be made more flexible.
+
+Decision
+--------
+Modify the AdditionalMetadata field validation logic to base decisions on the source of external products. In particular, have 
+a configuration object detailing the mapping between different product sources and the required additionalMetadata fields for
```
Comment (AfaqShuaib09) [inline decisions/0005-product-source-based-required-fields.rst:17]: nit: `AdditionalMetadata`*

**Review by AfaqShuaib09 — APPROVED** [review]

**Review by DawoudSheraz — COMMENTED** [review]
- More context under each heading is needed, especially how source plays an important part in this
- Make sure to list the alternatives considered for this work

File: decisions/0005-product-source-based-required-fields.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+During the recent influx of external products into the marketplace, a number of products were ingested into publisher which did
+not have all the AdditionalMetadata fields associated to them. These products were coming from different sources and thus had
+different expectations of the validation in AdditionalMetadata fields. As such our current publisher flow, which required all of the
+AdditionalMetadata fields to be present regardless of the product source, needed to be made more flexible to dynamically update its
+validation strategy depending on the particular products.
+
+Decision
+--------
+Modify the AdditionalMetadata field validation logic to base decisions on the source of external products. In particular, have 
+a configuration object detailing the mapping between different product sources and the required additionalMetadata fields for
```
Comment (DawoudSheraz) [inline decisions/0005-product-source-based-required-fields.rst:19]: Please add the format we used for configuration and the challenges we faced to map the config to environment var.

**Review by DawoudSheraz — COMMENTED** [review]
- Please add some more details to Decision & Future enhancements
- Instead of using "we" in the doc, please use third-person pronoun.

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**